test/src/test-cppapi-aggregates.cc

```diff
@@ -1860,6 +1865,167 @@ TEST_CASE_METHOD(
+
+        query.ptr()->query_->add_aggregator_to_default_channel(
+            "Sum",
+            std::make_shared<tiledb::sm::SumAggregator<T>>(
+                tiledb::sm::FieldInfo(
+                    "a1", false, CppAggregatesFx<T>::nullable_, 1)));
+
+        CppAggregatesFx<T>::set_ranges_and_condition_if_needed(
+            array, query, false);
+
+        // Set the data buffer for the aggregator.
+        uint64_t cell_size = sizeof(T);
+        std::vector<uint64_t> count(1);
+        std::vector<typename tiledb::sm::sum_type_data<T>::sum_type> sum(1);
+        std::vector<uint8_t> sum_validity(1);
+        std::vector<uint64_t> dim1(100);
+        std::vector<uint64_t> dim2(100);
+        std::vector<uint8_t> a1(100 * cell_size);
+        std::vector<uint8_t> a1_validity(100);
+        query.set_layout(layout);
+        uint64_t count_data_size = sizeof(uint64_t);
+        uint64_t sum_data_size =
+            sizeof(typename tiledb::sm::sum_type_data<T>::sum_type);
+        // TODO: Change to real CPPAPI.
+        CHECK(query.ptr()
+                  ->query_
+                  ->set_data_buffer("Count", count.data(), &count_data_size)
+                  .ok());
+        CHECK(query.ptr()
+                  ->query_->set_data_buffer("Sum", sum.data(), &sum_data_size)
+                  .ok());
+        uint64_t returned_validity_size = 1;
+        if (CppAggregatesFx<T>::nullable_) {
+          // TODO: Change to real CPPAPI. Use set_validity_buffer from the
+          // internal query directly because the CPPAPI doesn't know what is
+          // an aggregate and what the size of an aggregate should be.
+          CHECK(query.ptr()
+                    ->query_
+                    ->set_validity_buffer(
+                        "Sum", sum_validity.data(), &returned_validity_size)
+                    .ok());
+        }
+
+        // Here we run a few iterations until the query completes and update the
+        // buffers as we go.
+        uint64_t curr_elem = 0;
+        uint64_t num_elems = CppAggregatesFx<T>::dense_ ? 20 : 4;
+        for (uint64_t iter = 0; iter < 10; iter++) {
+          query.set_data_buffer("d1", dim1.data() + curr_elem, num_elems);
+          query.set_data_buffer("d2", dim2.data() + curr_elem, num_elems);
+          query.set_data_buffer(
+              "a1",
+              static_cast<void*>(a1.data() + curr_elem * cell_size),
+              num_elems);
+
+          if (CppAggregatesFx<T>::nullable_) {
+            query.set_validity_buffer(
+                "a1", a1_validity.data() + curr_elem, num_elems);
+          }
+
+          // Submit the query.
+          query.submit();
+
+          // Adjust current element count;
+          auto result_el = query.result_buffer_elements_nullable();
+          curr_elem += std::get<1>(result_el["d1"]);
+
+          // Stop on query completion.
+          if (query.query_status() == Query::Status::COMPLETE) {
+            break;
```

Comment:
Can we require that this breaks after a specific number of iterations? That would more robustly ensure that this test continues to do what we want in the future.